apps/prairielearn/src/lib/editors.js

```diff
@@ -82,134 +82,177 @@ export class Editor {
-                    '-c',
-                    `user.email="${this.user.uid}"`,
-                    'commit',
-                    '-m',
-                    this.commitMessage,
-                  ],
-                  {
-                    cwd: this.course.path,
-                    env: gitEnv,
-                  },
-                );
-              } catch (err) {
-                await cleanAndResetRepository(this.course, gitEnv, job);
-                throw err;
-              }
+        try {
+          await job.exec('git', ['add', ...this.pathsToAdd], {
+            cwd: this.course.path,
+            env: gitEnv,
+          });
+          await job.exec(
+            'git',
+            [
+              '-c',
+              `user.name="${this.user.name}"`,
+              '-c',
+              `user.email="${this.user.uid}"`,
+              'commit',
+              '-m',
+              this.commitMessage,
+            ],
+            {
+              cwd: this.course.path,
+              env: gitEnv,
+            },
+          );
+        } catch (err) {
+          await cleanAndResetRepository(this.course, gitEnv, job);
+          throw err;
+        }
 
-              try {
-                job.data.pushAttempted = true;
-
-                await job.exec('git', ['push'], {
-                  cwd: this.course.path,
-                  env: gitEnv,
-                });
-
-                // We'll look for this flag on the `editError` page to know if
-                // we need to display instructions to recover from a failed push.
-                job.data.pushSucceeded = true;
-              } finally {
-                // Regardless of whether we error, we'll do a clean and reset:
-                //
-                // If pushing succeeded, the clean will remove any empty directories
-                // that might have been left behind by operations like renames.
-                //
-                // If pushing errored, the reset will get us back to a known good state.
-                await cleanAndResetRepository(this.course, gitEnv, job);
-              }
+        if (!config.fileEditorUseGit) {
+          await syncCourseFromDisk(this.course, startGitHash, job);
+          return;
+        }
 
-              job.data.syncAttempted = true;
+        try {
+          await job.exec('git', ['add', ...this.pathsToAdd], {
+            cwd: this.course.path,
+            env: gitEnv,
+          });
+          await job.exec(
+            'git',
+            [
+              '-c',
+              `user.name="${this.user.name}"`,
+              '-c',
+              `user.email="${this.user.uid}"`,
+              'commit',
+              '-m',
+              this.commitMessage,
+            ],
+            {
+              cwd: this.course.path,
+              env: gitEnv,
+            },
+          );
+        } catch (err) {
+          await cleanAndResetRepository(this.course, gitEnv, job);
+          throw err;
+        }
 
```

Comment:
@mylesw27 I think something funky happened during merge conflict resolution. Did you mean to duplicate all this code?

Reply:
No, I did not. Removed in [a0985bb](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/a0985bb8f9b393c5d468b3fdfc5a1637597ec1fb).